### PR TITLE
[configgrpc] Remove deprecated functions/structs

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:**
Removes deprecated functions/structs

**Link to tracking Issue:** 
Related to https://github.com/open-telemetry/opentelemetry-collector/issues/9428
Related to https://github.com/open-telemetry/opentelemetry-collector/issues/9482
Closes https://github.com/open-telemetry/opentelemetry-collector/issues/9481

### CODE REVIEW

File: config/configgrpc/configgrpc.go [hunk 1]
```diff
@@ -190,7 +168,7 @@ func (gcs *ClientConfig) ToClientConn(ctx context.Context, host component.Host,
 		return nil, err
 	}
 	opts = append(opts, extraOpts...)
-	return grpc.DialContext(ctx, gcs.SanitizedEndpoint(), opts...)
+	return grpc.DialContext(ctx, gcs.Endpoint, opts...)
```
Comment (TylerHelmuth) [inline config/configgrpc/configgrpc.go:171]: This feels like a breaking change we shouldn't be doing.  Won't this affect users of the `otlpreceiver` how do something like:

```yaml
receivers:
  otlp:
    protocols:
      grpc:
        endpoint: https://my-address:4317
```

If the intent was only to hide this function publicly I can add this feature back privately.

**Comment by TylerHelmuth** [comment]
Requires https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31360

File: exporter/otlpexporter/config.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ type Config struct {
 }
 
 func (c *Config) Validate() error {
-	if c.SanitizedEndpoint() == "" {
```
Comment (TylerHelmuth) [inline exporter/otlpexporter/config.go:25]: SanitizedEndpoint is being used publicly here

**Comment by TylerHelmuth** [comment]
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31360 is ready for review

File: config/configgrpc/configgrpc.go (hunk 1 above)
Comment (bogdandrutu) [reply config/configgrpc/configgrpc.go]: Very confused about this comment. Please explain in the PR description

Comment (TylerHelmuth) [reply config/configgrpc/configgrpc.go]: @bogdandrutu I've removed all `SanitizedEndpoint` deprecations from this PR until we can discuss https://github.com/open-telemetry/opentelemetry-collector/issues/9482 more so you can disregard my comment in the context of this PR.

**Pull request merged**